### PR TITLE
add < to NamedTuple

### LINKED ISSUE #40142: Add < for NamedTuple
I think we should add:

```
<(a::NamedTuple{n}, b::NamedTuple{n}) where {n} = Tuple(a) < Tuple(b)
```

as now we fall back to `isless` which means that comparing `NamedTuple`s with `missing` values does not produce `missing`, so the following happens:
```
julia> (a=1,b=missing) <= (a=1, b=missing)
missing

julia> (a=1,b=missing) < (a=1, b=missing)
false

julia> (a=1,b=missing) > (a=1, b=missing)
false

julia> (a=1,b=missing) >= (a=1, b=missing)
missing

julia> (a=1,b=missing) == (a=1, b=missing)
missing
```

This is probably for 2.0 release, but maybe it should be considered a bug?

### PR DESCRIPTION
**Pull request by bkamins**
Fixes https://github.com/JuliaLang/julia/issues/40142

### CODE REVIEW

File: test/namedtuple.jl [hunk 1]
```diff
@@ -95,6 +95,14 @@ end
 @test !isless((a=2,b=1), (a=1,b=2))
 @test_throws MethodError isless((a=1,), (x=2,))
 
+@test (a=1,b=2) < (a=1,b=3)
+@test_broken (a=1,) < (a=1,b=2)
```
Comment (nalimilan) [inline test/namedtuple.jl:99]: This should probably throw an error? Of course that's not possible until 2.0, but maybe a deprecation should be printed?

Comment (bkamins) [reply test/namedtuple.jl:99]: This is an error. I thought the `@test_broken` above was intentional. I will change this.

Comment (bkamins) [reply test/namedtuple.jl:99]: The relevant TODO is https://github.com/JuliaLang/julia/pull/40147/files#diff-8bd5cf0ce6c0ef236a5ae91851437fc8197d8f56c70b2b5dedd73ef55bb18e17L198, but personally I agree we should not accept prefixes.

File: test/namedtuple.jl [hunk 2]
```diff
@@ -90,11 +90,19 @@ end
 @test Tuple((a=1, b=2, c=3)) == (1, 2, 3)
 
 @test isless((a=1,b=2), (a=1,b=3))
-@test_broken isless((a=1,), (a=1,b=2))
+@test MethodError isless((a=1,), (a=1,b=2))
```
Comment (bkamins) [inline test/namedtuple.jl:93]:
```suggestion
@test_throws MethodError isless((a=1,), (a=1,b=2))
```

File: test/namedtuple.jl [hunk 3]
```diff
@@ -90,11 +90,19 @@ end
...
 @test !isless((a=1,b=2), (a=1,b=2))
 @test !isless((a=2,b=1), (a=1,b=2))
 @test_throws MethodError isless((a=1,), (x=2,))
 
+@test (a=1,b=2) < (a=1,b=3)
+@test MethodError (a=1,) < (a=1,b=2)
```
Comment (bkamins) [inline test/namedtuple.jl:99]:
```suggestion
@test_throws MethodError (a=1,) < (a=1,b=2)
```

File: test/namedtuple.jl [hunk 4]
```diff
@@ -90,11 +90,19 @@ end
...
 @test !isless((a=2,b=1), (a=1,b=2))
 @test_throws MethodError isless((a=1,), (x=2,))
 
+@test (a=1,b=2) < (a=1,b=3)
+@test_throws MethodError (a=1,) < (a=1,b=2)
+@test !((a=1,b=2) < (a=1,b=2))
+@test !((a=2,b=1) < (a=1,b=2))
+@test_throws MethodError (a=1,) < (x=2,)
+@test !((a=-0.0) < (a=0.0))
+@test ismissing((a=missing) < (a=missing))
```
Comment (nalimilan) [inline test/namedtuple.jl:104]:
```suggestion
@test !((a=-0.0,) < (a=0.0,))
@test ismissing((a=missing,) < (a=1,))
@test ismissing((a=missing,) < (a=missing,))
```

**Pull request merged**